genshin/client/components/base.py

```diff
@@ -356,6 +356,58 @@ async def request_webstatic(
 
         return data
 
+    def _get_ds_headers(
+        self,
+        region: types.Region,
+        data: typing.Any = None,
+        params: typing.Optional[typing.Mapping[str, typing.Any]] = None,
+        lang: typing.Optional[str] = None,
+    ) -> typing.Dict[str, typing.Any]:
+        if region == types.Region.OVERSEAS:
+            ds_headers = {
+                "x-rpc-app_version": "1.5.0",
+                "x-rpc-client_type": "4",
+                "x-rpc-language": lang,
+                "ds": ds.generate_dynamic_secret(),
+            }
+        elif region == types.Region.CHINESE:
+            ds_headers = {
+                "x-rpc-app_version": "2.11.1",
+                "x-rpc-client_type": "5",
```

Comment:
Can you confirm that the value of the `client_type` is 5 and not 4. I see this value of 4 in the console.

Reply:
Well this is from your old code

Reply:
I'm pretty sure I took it from someone else. Everything related to Chinese endpoints is either contributed by someone else or taken from other projects.

Reply:
Lol then keep it 5

Reply:
If you are sure that it won't break anything. I thought it needed a newer DS version and all that.

Reply:
It's midnight and my computer is off, so I can't confirm if it works. Also I found from other projects that the DS_SALT used is not the same when the client_type is different. If it doesn't work, I think he really needs a new DS method.

Reply:
I guess it might be a good idea to implement the same thing as is in here then

https://github.com/Le-niao/Yunzai-Bot/blob/07f39c89bd2603b447ba924aea43e55c2b6d6549/plugins/genshin/model/mys/mysApi.js#L230-L307

Reply:
This DS_SALT works , for now.

But this salt version is kinda old.

Also, I guess client_type is related to platform, mobile =4 and web =5 something like that.

Interesting, seems like he uses different ds for daily reward. And he add ```x-rpc-device_id```,```x-rpc-platform```,```x-rpc-device_model```,```x-rpc-device_name``` headers for daily reward requests. Looks like a good way to avoid geetest?

Reply:
Yes, client_type is related to platform.
ios = 1
android = 2
pc web = 4
mobile web = 5
https://github.com/y1ndan/genshinhelper2/blob/5402ffbe76885432d387e3cf04bb9d1dcefb9075/genshinhelper/utils.py#L163-L202

According to my logs in a production environment, building the full request header is the way to avoid geetest.

Reply:
Does the request ip difference also triggers geetest?

Reply:
Alright, sounds good